simulation/sub8_gazebo/nodes/pinger_sim.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env python
+import rospy
+from gazebo_msgs.srv import GetModelState
+from sub8_msgs.srv import Sonar
+from sub8_sonar import EchoLocator
+from sub8_ros_tools import msg_helpers
+from sub8_ros_tools.geometry_helpers import rotate_vect_by_quat
+import tf
+
+import numpy as np
+
+
+class Pinger():
+    def __init__(self, wave_propagation_speed, precision=14):
+        self.get_model = rospy.ServiceProxy('/gazebo/get_model_state', GetModelState)
+
+        self.wave_propagation_speed = wave_propagation_speed
+        self.precision = precision
+
+        self.hydrophone_board_tf = np.array([0, 0, 0])
+        self.hydro_loc =  {'hydro0': {'x': 0,'y': 0,'z': 0},
+                           'hydro1': {'x': -0.0254, 'y': 0, 'z': 0.0254},
+                           'hydro2': {'x': 0.0254, 'y': 0, 'z': 0},
+                           'hydro3': {'x': 0, 'y': -0.0254, 'z': 0} }
+
+        self.echo_locator = EchoLocator(self.hydro_loc, wave_propagation_speed)
+
+        # Wait for all the models and such to spawn.
+        rospy.Service('/sonar/get_pinger_pulse', Sonar, self.request_data)
+
+        rospy.sleep(3)
+        print self.request_data(None)
```

Comment:
When testing I would constantly restart the node - this made it easier to test quickly. I forgot to remove a bunch of the debugging stuff